### PR TITLE
Added Debug Shapes Layer to Thunderscope

### PR DESCRIPTION
**Pull request by nimazareian**
### Description
Adding a new layer to Thunderscope to allow visualizing arbitrary Stadium, Circle, or Polygon shapes from any where in our AI. This could be helpful for quickly visualizing arbitrary shapes that we use for evaluation in our FSMs or motion planner. 

_Note_: The layer is hidden by default, so you will need to manually unhide it to see any values visualized.

image

### Testing Done
Tried adding some dynamic shapes to our AI code and it seems to work without any error. The code I used:

```cpp
    static double static_radius = 0.1;
    static_radius += 0.001;
    LOG(VISUALIZE) << *createDebugShapesMap({
        {"circle_name", *createShapeProto(Circle(Point(1,0), static_radius))},
        {"stadium_name", *createShapeProto(Stadium(Point(-1,0), Point(-3,0), 0.5))},
        {"polygon_name", *createShapeProto(Polygon({Point(-2, 0), Point(-3, 0), Point(-2.5, 1)}))}
    });
```

### Resolved Issues
Been on my mind for a while. Didn't have an issue :sweat_smile: 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [X] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [X] **Remove all commented out code**
- [X] **Remove extra print statements**: for example, those just used for testing
- [X] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

**Review by williamckha — DISMISSED** [review]
LGTM!

File: src/proto/message_translation/tbots_protobuf.h [hunk 1]
```diff
@@ -126,6 +126,29 @@ std::unique_ptr<TbotsProto::NamedValue> createNamedValue(const std::string name,
...
+ *
+ * @return The unique_ptr to a TbotsProto::DebugShapesMap proto containing data with
+ *        specified names and shapes
+ */
+std::unique_ptr<TbotsProto::DebugShapesMap> createDebugShapesMap(
+    const std::map<std::string, TbotsProto::Shape>& named_shapes);
+
+std::unique_ptr<TbotsProto::Shape> createShapeProto(const Circle& circle);
+std::unique_ptr<TbotsProto::Shape> createShapeProto(const Polygon& circle);
+std::unique_ptr<TbotsProto::Shape> createShapeProto(const Stadium& circle);
```
Comment (itsarune) [inline src/proto/message_translation/tbots_protobuf.h]: nit: all the parameters all called circle

**Review by itsarune — COMMENTED** [review]
looks good! I used it to visualize passes and it's quite nice.

In the future, it would be nice to control the display colour via the DebugShape proto

good to merge, pending one nit

File: src/proto/message_translation/tbots_protobuf.h (hunk 1 above)
Comment (nimazareian) [reply src/proto/message_translation/tbots_protobuf.h]: Updated!

**Comment by nimazareian** [comment]
> In the future, it would be nice to control the display colour via the DebugShape proto

Ya, that's a good idea. I wanted to keep the interface relatively simple so I initially didn't look into that. I just tried seeing what it takes and it might require some rewrite since currently we set the color of the shapes all at once, so having different colors per distinct shape might be a little complex.

I chose purple since we're not really using it right now, but I'd be happy to change the default to something else for now if you want.

**Review by williamckha — APPROVED** [review]

**Review by sauravbanna — APPROVED** [review]
LGTM

**Review by itsarune — APPROVED** [review]
LGTM!

**Pull request merged**